controllers/volumereplicationgroup_controller.go

```diff
@@ -364,3 +360,126 @@ func (v *VolumeReplicationGroupReconciler) SetupWithManager(mgr ctrl.Manager) er
 		Owns(&ramendrv1alpha1.VolumeReplicationGroup{}).
 		Complete(v)
 }
+
+// Upload the PV of the given PVC to object store, if the
+// VRG has been configured with an s3Endpoint.
+// - Return an error if S3Endpoint is configured but the PV
+//   could not be uploaded for any reason.
+// - If S3Endpoint is not configured, the user does not want
+//   VRG to backup/replicate the PV to an object store and has
+// 	 other means to replicate PV metadata and is likely using
+//   the VRG as a single control point to manipulate replication
+// 	 of a group of VRs and thus a group of PVs.  However, log
+//   a warning once per VRG
+func (v *VolumeReplicationGroupReconciler) uploadPVofPVC(
+	ctx context.Context,
+	volRepGroup *ramendrv1alpha1.VolumeReplicationGroup,
+	pvc *corev1.PersistentVolumeClaim) error {
+	vrgName := volRepGroup.Name
+
+	s3Endpoint := volRepGroup.Spec.S3Endpoint
+	if s3EndpointIsUnconfiguredOrInvalid(s3Endpoint, vrgName) {
+		return nil
```

Comment:
Because we've logged an error already, especially for the case of empty S3Endpoint.

Reply:
Right, but if you return `nil` from uploadPVofPVC, then the caller of `uploadPVofPVC` will expect that everything is well, but in reality, things didn't go well because of misconfiguration or whatever.

Reply:
True, I've fixed it now.